### PR TITLE
[IMP] pos_self_order: change authorisation token

### PR DESCRIPTION
**Pull request by davidmonnom**
Previously, access_token were linked to tables. These tables were linked
to a floor_plan, which could be linked to several pos_configs.

The only way to access the self-order was to obtain a valid table
access_token. This behaviour was not correct because sometimes we would
allow table selection directly in the interface or commands without
a table.

Now, access_token is managed by pos_config. When a user has this token,
they can place commands and select the table they want if the option is
enabled.

### CODE REVIEW

File: addons/pos_self_order/controllers/orders.py [hunk 1]
```diff
@@ -96,21 +90,28 @@ def get_order_taxes(self, order, pos_config_id):
...
-        pos_config_sudo = get_pos_config_sudo(pos_config_id)
+
+        table_sudo = request.env["restaurant.table"].sudo().search([('identifier', '=', table_identifier)], limit=1)
+        pos_config_sudo = request.env['pos.config'].sudo().search([('access_token', '=', access_token)], limit=1)
+
+        if not pos_config_sudo or not pos_config_sudo.self_order_table_mode or not pos_config_sudo.has_active_session:
+            raise Unauthorized("Invalid access token")
+
+        if not table_sudo:
+            raise NotFound("Table not found")
```
Comment (caburj) [inline addons/pos_self_order/controllers/orders.py:104]: This logic seems to be duplicated. You can extract.

File: addons/pos_self_order/demo/custom_link_data.xml
Comment (caburj) [inline addons/pos_self_order/demo/custom_link_data.xml:1]: DAMA is not in favor of removing this custom link data. It's possible that users already setup there self-order setting and expects the buttons they have configured. Let's not hard-code the default link to the menu.

File: addons/pos_self_order/static/src/pages/landing_page/landing_page.xml [hunk 2]
```diff
@@ -15,7 +15,13 @@
                         <t t-esc="link.name"/>
                     </a>
                 </t>
-                <t t-if="selfOrder.table and selfOrder.self_order_mode === 'each'">
+                <a
+                    type="button"
+                    t-on-click="() => this.router.navigate('productList')"
+                    class="btn btn-primary py-3 my-2">
+                    View Menu
+                </a>
```
Comment (caburj) [inline addons/pos_self_order/static/src/pages/landing_page/landing_page.xml:23]: As per the other comment on the removal of the custom link data, we should not hardcode this link. We should respect the customers' setup.

File: addons/pos_self_order/static/src/components/popup_table/popup_table.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            acc[table.floor_name] = {
+                id: table.floor_id[0],
+                name: table.floor_id[1],
+                tables: this.tables.filter((t) => t.floor_id[0] === table.floor_id[0]),
+            };
+
+            return acc;
+        }, {});
+
+        return Object.values(floors);
```
Comment (caburj) [inline addons/pos_self_order/static/src/components/popup_table/popup_table.js:43]:
```suggestion
        const groupedFloors = groupBy(this.tables, (t) => t.floor_id[0]);
        return Object.entries(groupedFloors).map(([floorId, tables]) => ({
              id: floorId,
              name: tables[0].floor_id[1],
              tables
        }));
```

**Review by caburj — COMMENTED** [review]
Left some comments. Will continue tomorrow.

File: addons/pos_self_order/models/pos_restaurant.py [hunk 4]
```diff
@@ -23,45 +23,35 @@ def _get_data_for_qr_codes_page(self, url: Callable):
 class RestaurantTable(models.Model):
     _inherit = "restaurant.table"
 
-    access_token = fields.Char(
+    identifier = fields.Char(
```
[the remark this answers is not in the document]
Comment (davidmonnom) [reply addons/pos_self_order/models/pos_restaurant.py:26]: I think we'll lose the data, but we don't need to keep it anyway.

File: addons/pos_self_order/demo/custom_link_data.xml
Comment (davidmonnom) [reply addons/pos_self_order/demo/custom_link_data.xml:1]: I'll be discussing this on Monday to make sure everything is understood.

File: addons/pos_self_order/models/pos_restaurant.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (caburj) [reply addons/pos_self_order/models/pos_restaurant.py:26]: I'm wrong. It seems that `access_token` is introduced in this current master. So you don't need a migration script.

**Review by caburj — APPROVED** [review]
robodoo r+

**Pull request closed without merging**